db/create.go

```diff
@@ -0,0 +1,156 @@
+// SQLite Caching and Cross-Exploit Database
+//
+// The db package contains the logic to handle a user provided SQLite DB in order
+// to store results and cache HTTP responses. This has a few useful benefits:
+//
+//  1. When scanning with hundreds of go-exploit implementations, the user significantly
+//     cuts down on network requests (therefore speeding up scanning), both from the
+//     verified results being cached (you only have to verify a target is Confluence once)
+//     and from the cached HTTP responses.
+//  2. The result is a useful asset database containing IP, port, installed software, and
+//     versions.
+//  3. The database can be reused with a go-exploit and generate no network traffic (assuming
+//     you aren't doing the exploitation stage). That is very interesting when, for example,
+//     you wrote a version scanner for CVE-2024-31982, scanned a customer host that was patched,
+//     but then CVE-2024-31983 is released the next day. You can essentially rescan the cached
+//     version of their system with your new CVE scanner.
+//
+// Mostly this package should be totally transparent to users of the framework. The only direct
+// interface, currently, should be calls to HTTPGetCache.
+package db
+
+import (
+	"database/sql"
+	"time"
+
+	"github.com/vulncheck-oss/go-exploit/output"
+	// pure go sqlite3 driver.
+	_ "modernc.org/sqlite"
+)
+
+// GlobalSQLHandle is a handle to the SQLite DB for handling cross-exploit data sharing.
+var GlobalSQLHandle *sql.DB
+
+// GlobalHTTPRespCacheLimit is the maximum size of an HTTP body that we will attempt to cache.
+var GlobalHTTPRespCacheLimit int
+
+const (
+	schemaVersion = "1.0.0"
```

Comment:
Do we have plans on how to handle migrations or should that just be a future us problem?